src/ctapipe/instrument/optics.py

```diff
@@ -263,3 +269,161 @@ def __repr__(self):
 
     def __str__(self):
         return self.name
+
+
+class PSFModel:
+    def __init__(self, **kwargs):
+        """
+        Base component to describe image distortion due to the optics of the different cameras.
+        """
+
+    @classmethod
+    def subclass_from_name(cls, name, **kwargs):
+        """
+        Obtain an instance of a subclass via its name
+
+        Parameters
+        ----------
+        name : str
+            Name of the subclass to obtain
+
+        Returns
+        -------
+        Instance
+            Instance of subclass to this class
+        """
+
+        subclasses = {base.__name__: base for base in non_abstract_children(cls)}
+
+        requested_subclass = subclasses[name]
+        return requested_subclass(**kwargs)
+
+    @abstractmethod
+    def pdf(self, *args):
+        pass
+
+    @abstractmethod
+    def update_location(self, *args):
+        pass
+
+    @abstractmethod
+    def update_model_parameters(self, *args):
+        pass
+
+
+class ComaModel(PSFModel):
+    """
+    PSF model, describing pure coma aberrations PSF effect
+    """
+
+    def __init__(
+        self,
+        asymmetry_params=[0.49244797, 9.23573115, 0.15216096],
+        radial_scale_params=[0.01409259, 0.02947208, 0.06000271, -0.02969355],
+        az_scale_params=[0.24271557, 7.5511501, 0.02037972],
+    ):
+        """
+        PSF model, describing purely the effect of coma aberration on the PSF
+
+        Parameters
+        ----------
+        asymmetry_params: list
+            Parameters describing the dependency of the asymmetry of the psf on the distance to the center of the camera
+        radial_scale_params : list
+            Parameters describing the dependency of the radial scale on the distance to the center of the camera
+        az_scale_params : list
+            Parameters describing the dependency of the azimuthal scale scale on the distance to the center of the camera
+        """
+
+        self.asymmetry_params = asymmetry_params
+        self.radial_scale_params = radial_scale_params
+        self.az_scale_params = az_scale_params
+
+        self.update_location(0.0, 0.0)
+
+    def k_func(self, x):
+        return (
+            1
+            - self.asymmetry_params[0] * np.tanh(self.asymmetry_params[1] * x)
+            - self.asymmetry_params[2] * x
+        )
+
+    def sr_func(self, x):
+        return (
```

Comment:
This looks like it should use 

https://numpy.org/doc/stable/reference/generated/numpy.polyval.html